arangod/Cluster/DBServerAgencySync.cpp

```diff
@@ -52,13 +59,75 @@ void DBServerAgencySync::work() {
   _heartbeat->dispatchedJobResult(result);
 }
 
+Result getLocalCollections(VPackBuilder& collections) {
+
+  using namespace arangodb::basics;
+  Result result;
+  DatabaseFeature* dbfeature = nullptr;
+
+  try {
+    dbfeature = ApplicationServer::getFeature<DatabaseFeature>("Database");
+  } catch (...) {}
+
+  if (dbfeature == nullptr) {
+    LOG_TOPIC(ERR, Logger::HEARTBEAT) << "Failed to get feature database";
+    return Result(TRI_ERROR_INTERNAL, "Failed to get feature database");
+  }
+  
+  VPackObjectBuilder c(&collections);
+  for (auto const& database : Databases::list()) {
+    auto vocbase = dbfeature->lookupDatabase(database);
+    if (vocbase == nullptr) {
+      continue;
+    }
+
+    try {
+      DatabaseGuard guard(*vocbase);      
```

Comment:
This looks like a fast guard, then unguard.  Is that intentional?

Reply:
fixed